src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -41,26 +41,7 @@ public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> sourc
                 return new ListPartition<TSource>(sourceList, count, int.MaxValue);
             }
 
-            return SkipIterator(source, count);
-        }
-
-        private static IEnumerable<TSource> SkipIterator<TSource>(IEnumerable<TSource> source, int count)
-        {
-            using (IEnumerator<TSource> e = source.GetEnumerator())
-            {
-                while (count > 0 && e.MoveNext())
-                {
-                    count--;
-                }
-
-                if (count <= 0)
-                {
-                    while (e.MoveNext())
-                    {
-                        yield return e.Current;
-                    }
-                }
-            }
+            return new EnumerablePartition<TSource>(source, count, int.MaxValue);
```

Comment:
I can change this to use `long` instead. It's unlikely that an enumerable that comes close to `long.MaxValue` elements can be iterated in a reasonable amount of time.